inc/Editor/BlockManagement/ConfigRegistry.php

```diff
@@ -23,15 +23,57 @@ class ConfigRegistry {
 	public const RENDER_QUERY_KEY = 'render_query';
 	public const SELECTION_QUERIES_KEY = 'selection_queries';
 	public const DISPLAY_QUERY_KEY = 'display';
+	public const DISPLAY_QUERIES_KEY = 'display_queries';
 	public const LIST_QUERY_KEY = 'list';
 	public const SEARCH_QUERY_KEY = 'search';
+	public const QUERIES_KEY = 'queries';
 
 	public static function init( ?LoggerInterface $logger = null ): void {
 		self::$logger = $logger ?? new Logger();
 		ConfigStore::init( self::$logger );
 	}
 
+	private static function migrate_block_config( array $block_config = [] ): array|WP_Error {
+		// Nothing to migrate, return the block config as is.
+		if ( isset( $block_config[ self::QUERIES_KEY ] ) ) {
+			return $block_config;
+		}
+
+		// if render_query is not set, error out.
+		if ( ! isset( $block_config['render_query'] ) ) {
```

Comment:
I'm confused, I thought we were moving away from `render_query` towards `queries`.

Reply:
Indeed, this is the second migration system targets the entire block config instead of just a single query.